### ISSUE TITLE
CATE with continuous treatment and categorical outcome 

### ISSUE THREAD

**Issue by hadi-gharibi** [DoubleML/doubleml-for-py#203]
I was trying to use DML for continuous treatment (price) and binary outcome(churn). Based on the docs, its not possible to use any of these techniques to this case. Is there any way to adjust any of these algorithms for this setup? If there is a paper that I could develop and add to this library I'm down to it as well.

**Comment by SvenKlaassen** [comment]
Yes, I think DML would be suitable to estimate dose-response relationships. 
But currently, we have no model which specifically enforces the combination of continous treatment and binary outcomes.
I think reference [Double/debiased machine learning for logistic partially linear model](https://academic.oup.com/ectj/article-abstract/24/3/559/6296639) posted by @PhilippBach seems really helpful for this setting.

**Comment by hadi-gharibi** [comment]
Thanks for the response @PhilippBach  and @SvenKlaassen 
Since I need this, I have to make it work, one way or another. If you think it would fit in DML library setup, I can give it a shot. 
I can expand the `DoubleMLPLR` to support continuous treatment and binary outcomes, but I think It needs a different `_check_learner` and lots of if and else here and there. I personally don't think it would be a good approach.

Another way is to have a separate class to support that, like `DoubleMLLPLR` for the logistic partially linear model. it would work just fine.

Yet another idea could be to keep the `DoubleMLPLR` as a strategy class that picks the correct approach( linear or logistic class) based on the dataset.

**Comment by chelsea-lee-wbd** [comment]
This is not completely related to this topic but has some overlap. 

On the topic of binary outcome, `DoubleMLPLR` is mentioned but I noticed that `ml_l` is treated as a continuous outcome from this documentation:[](https://docs.doubleml.org/stable/api/generated/doubleml.DoubleMLPLR.html )Looking at the code, it looks like the predictions for the binary outcome uses class labels instead of probabilities (`pred()` vs `pred_proba()`). 

1. Is the above information correct? Please let me know if I missed something. I was looking at whether `DoubleMLIRM` is better suited instead
2. If 1) is true, are there implications to be concerned about when it comes to the residuals and the resulting quality of the ATE? My suspicion is that there'd be more bias

**Comment by chelsealee14** [comment]
This is not completely related to this topic but has some overlap. 

On the topic of binary outcome, `DoubleMLPLR` is mentioned but I noticed that `ml_l` is treated as a continuous outcome from this documentation:[](https://docs.doubleml.org/stable/api/generated/doubleml.DoubleMLPLR.html )Looking at the code, it looks like the predictions for the binary outcome uses class labels instead of probabilities (`pred()` vs `pred_proba()`). 

1. Is the above information correct? Please let me know if I missed something. I was looking at whether `DoubleMLIRM` is better suited instead
2. If 1) is true, are there implications to be concerned about when it comes to the residuals and the resulting quality of the ATE? My suspicion is that there'd be more bias

**Comment by PhilippBach** [comment]
Hi @chelsealee14 

thanks for your message. I'm not 100% if I get your point here, as this issue is related to CATEs 

`DoubleMLPLR` allows for continuous outcome only (as of now), see 

https://github.com/DoubleML/doubleml-for-py/blob/31f738842910cf95212a7d702fe9c45707ec0ce0/doubleml/plm/plr.py#L111

The linearity in the PLR model is a general challenge to predicting probabilities and binary outcomes. I don't remember exactly, but I think we tested some simulations for binary $Y$ with a `DoubleMLPLR` and a binary outcome and classification learner `ml_l` and it worked quite well. This would not be covered by theory, but could be a practical solution. Feel free to change that line of code above (setting `classifier = True` should be more or less enough)  and try it out yourself. The `DoubleMLIRM` model does not impose such an assumption and should work for binary outcomes too.

I hope this helps!
